provisioning/README.md

```diff
@@ -0,0 +1,81 @@
+# Mbed Crypto factory device provisioning example
+
+This example demonstrates how to install keys into a device during a
+hypothetical board assembly stage. Your device comprises a microcontroller, a
+secure element (e.g. ATECC608A), and a PCB to connect and hold everything
+together. When this README refers to "device", we mean the entire board
+assembly, not just the microcontroller or the secure element.
+
+Let's say the secure element manufacturer shipped you a secure element with a
+device private key already inside. This type of "pre-provisioned" secure
+element is what you'll use as a part during the device assembly process. You've
+taken this secure element, paired it with a nice Cortex-M microcontroller on a
+board to make a device. Now, you'd like to be able to use this key from Mbed
+Crypto. How does Mbed Crypto know the key exists? We can tell Mbed Crypto by
+running a factory device provisionining application, which is typically
+separate from the primary application and may be run as a final device assembly
+step.
+
+
+### Installing keys during factory device provisioning
+
+Mbed Crypto provides three ways to install secure element keys into your
+device.
+
+- Register a pre-existing secure element key
+- Generate a new key within a secure element
+- Import a pre-existing key to a secure element (not all secure elements
+  support this method)
+
+This example demonstrates the first two methods.
+
+
+#### Register a pre-existing secure element key
+
+For registering keys already present within a secure element, Mbed Crypto
+provides a function to inform Mbed Crypto about the existence of the key:
+`mbedtls_psa_register_se_key()`. This function adds the necessary metadata to
+persistent storage so that the secure element keys can be used by an
+application.
+
+With `psa_set_key_slot_number()`, you can specify which physical secure element
+slot the key is in. This function operates on a key attributes structure. Fill
+in any other necessary attributes and then call `mbedtls_psa_register_se_key()`
```

Comment:
“any other necessary attributes” makes it seem that just setting the slot number can be enough, but it never is. Lifetime and key id are always necessary, usage policy most of the time. Type and size may or may not be necessary depending on whether the hardware already has this information (e.g. in the slot configuration). Either change it to “other necessary attributes” or list the attributes.